src/Arcus.Security.Core/CompositeSecretProvider.cs

```diff
@@ -161,6 +173,7 @@ public async Task InvalidateSecretAsync(string secretName)
                 throw new SecretNotFoundException(secretName, noRegisteredException);
             }
 
+            var criticalExceptions = new Collection<Exception>();
```

Comment:
By collecting the exceptions this way, we'll loose track of the actual stacktrace information but the exception was logged with stracktrace also, so that's maybe not a problem.